environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java

```diff
@@ -230,11 +231,10 @@ private CreateFreeIpaRequest createFreeIpaRequest(EnvironmentDto environment) {
     }
 
     private void setTags(EnvironmentDto environment, CreateFreeIpaRequest createFreeIpaRequest) {
-        Map<String, String> userDefinedTags = environment.getTags().getUserDefinedTags();
-        if (userDefinedTags == null) {
-            userDefinedTags = new HashMap<>();
-        }
-        createFreeIpaRequest.setTags(userDefinedTags);
+        Optional.ofNullable(environment.getTags())
+                .map(EnvironmentTags::getUserDefinedTags)
+                .map(tags -> new TagsRequest(tags.getAll()))
+                .ifPresent(createFreeIpaRequest::setTags);
```

Comment:
it won't set the empty Tags if there was no `userDefinedTags` present

Reply:
the request has an empty tags default value, so it is not necessary